edb/pgsql/compiler/relgen.py

```diff
@@ -3898,6 +3899,89 @@ def process_set_as_fts_search(
         ir_set, obj_arg_offset=2, inner_cb=cb, ctx=ctx)
 
 
+@_special_case('ext::ai::search')
+def process_set_as_ext_ai_search(
+    ir_set: irast.SetE[irast.Call], *, ctx: context.CompilerContextLevel
+) -> SetRVars:
+    cb = _ext_ai_search_inner_pgvector
+    return _process_set_as_object_search(
+        ir_set, obj_arg_offset=0, inner_cb=cb, ctx=ctx)
+
+
+def _ext_ai_search_inner_pgvector(
+    call: irast.Call,
+    obj_id: irast.PathId,
+    args_pg: list[pgast.BaseExpr],
+    _ctx: context.CompilerContextLevel,
+    newctx: context.CompilerContextLevel,
+    _inner_ctx: context.CompilerContextLevel,
```

Comment:
This one doesn't need to be addressed yet since it's really an FTS thing, but having three different contexts is quite complex and needs to be explained and given likely better names. (I think it's because the FTS stuff is picky about where stuff goes?)

Reply:
@aljazerzen ^